Gem/Code/Source/Automation/ScriptManager.cpp

```diff
@@ -1635,15 +1666,10 @@ namespace AtomSampleViewer
             });
     }
 
-    void ScriptManager::OnCaptureFinished(uint32_t frameCaptureId, AZ::Render::FrameCaptureResult result, const AZStd::string &info)
+    void ScriptManager::OnFrameCaptureFinished(AZ::Render::FrameCaptureResult result, const AZStd::string &info)
     {
-         // ignore captures that are not triggered by the script manager
-        if (m_frameCaptureId == AZ::Render::FrameCaptureRequests::s_InvalidFrameCaptureId || frameCaptureId != m_frameCaptureId)
-        {
-            return;
-        }
         m_isCapturePending = false;
-        m_frameCaptureId = AZ::Render::FrameCaptureRequests::s_InvalidFrameCaptureId;
+        m_frameCaptureId = AZ::Render::InvalidFrameCaptureId;
         AZ::Render::FrameCaptureNotificationBus::Handler::BusDisconnect();
```

Comment:
Now it checks `frameCaptureId != AZ::Render::InvalidFrameCaptureId` before connecting to the bus, we can skip disconnecting here as well.

Reply:
It's more important to disconnect now as FrameCaptureId's are re-used. Additionally we would need further changes for the ScriptManager to be able to handle multiple active frame capture notification connections.